### PR TITLE
Http/make request public

### PR DESCRIPTION
**Pull request by vhvb1989**
Update the http request to be public (remove underscore)

fixes: https://github.com/Azure/azure-sdk-for-c/issues/950

### CODE REVIEW

File: sdk/inc/azure/core/az_http_transport.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ AZ_INLINE az_http_method az_http_method_patch() { return AZ_SPAN_FROM_STR("PATCH
 typedef az_span _az_http_request_headers;
 
 /**
- * @brief _az_http_request is an internal structure used to perform an HTTP request.
+ * @brief az_http_request is an structure used to perform an HTTP request.
```
Comment (JeffreyRichter) [inline sdk/inc/azure/core/az_http_transport.h:53]: We should add docs saying that this type should only be used by someone implementing an HTTP adapter.

Comment (vhvb1989) [reply sdk/inc/azure/core/az_http_transport.h:53]: We do.
The header file is called "az_http_transport.h". Meant to be included by someone implementing a transport adapter.
Also in the top of the file we mentioned it

Comment (ahsonkhan) [inline sdk/inc/azure/core/az_http_transport.h:53]:
```suggestion
 * @brief Structure used to represent an HTTP request.
```

File: sdk/inc/azure/core/az_http_transport.h [hunk 2]
```diff
@@ -80,7 +80,7 @@ typedef struct _az_http_policy _az_http_policy;
 
 /**
  * @brief Defines the callback signature of a policy process which should receive an
- * _az_http_policy, options reference (as void *), an _az_http_request and az_http_response.
+ * _az_http_policy, options reference (as void *), an az_http_request and az_http_response.
```
Comment (ahsonkhan) [inline sdk/inc/azure/core/az_http_transport.h:83]:
```suggestion
 * #_az_http_policy, options reference (as void *), an #az_http_request and #az_http_response.
```

File: sdk/src/azure/platform/az_curl.c [hunk 3]
```diff
@@ -305,8 +305,7 @@ static AZ_NODISCARD az_result _az_http_client_curl_send_get_request(CURL* ref_cu
 /**
  * handles DELETE request
  */
-static AZ_NODISCARD az_result
-_az_http_client_curl_send_delete_request(CURL* ref_curl)
+static AZ_NODISCARD az_result _az_http_client_curl_send_delete_request(CURL* ref_curl)
```
Comment (ahsonkhan) [inline sdk/src/azure/platform/az_curl.c:308]: Fyi @antkmsft - see example of potential merge conflict introduced because of unrelated change in https://github.com/Azure/azure-sdk-for-c/pull/988 :)
